vg/draw/canvas.go

```diff
@@ -256,39 +286,23 @@ func New(c vg.CanvasSizer) Canvas {
 }
 
 // NewFormattedCanvas creates a new vg.CanvasWriterTo with the specified
-// image format.
+// image format. Supported formats need to be registered by importing one or
+// more of the following packages:
 //
-// Supported formats are:
-//
-//  eps, jpg|jpeg, pdf, png, svg, tex and tif|tiff.
+//     gonum.org/v1/plot/vg/vgeps // provides eps
+//     gonum.org/v1/plot/vg/vgimg // provides png, jpg|jpeg, tif|tiff
+//     gonum.org/v1/plot/vg/vgpdf // provides pdf
+//     gonum.org/v1/plot/vg/vgsvg // provides svg
+//     gonum.org/v1/plot/vg/vgtex // provides tex
 func NewFormattedCanvas(w, h vg.Length, format string) (vg.CanvasWriterTo, error) {
-	var c vg.CanvasWriterTo
-	switch format {
-	case "eps":
-		c = vgeps.New(w, h)
-
-	case "jpg", "jpeg":
-		c = vgimg.JpegCanvas{Canvas: vgimg.New(w, h)}
-
-	case "pdf":
-		c = vgpdf.New(w, h)
-
-	case "png":
-		c = vgimg.PngCanvas{Canvas: vgimg.New(w, h)}
-
-	case "svg":
-		c = vgsvg.New(w, h)
-
-	case "tex":
-		c = vgtex.NewDocument(w, h)
-
-	case "tif", "tiff":
-		c = vgimg.TiffCanvas{Canvas: vgimg.New(w, h)}
-
-	default:
-		return nil, fmt.Errorf("unsupported format: %q", format)
+	formats.Lock()
+	defer formats.Unlock()
+	for name, fn := range formats.m {
+		if format == name {
+			return fn(w, h), nil
+		}
 	}
```

Comment:
Factor this out to a helper that returns the `fn`?

Reply:
it's only needed once.
I'd keep it like as it is (well, I'd err on using the `!=` + `continue` form that keeps the col-count lower and the "indentation arrow of death" at bay, but ok.)

Reply:
OK